service/src/main/java/bio/terra/workspace/service/resource/controlled/cloud/azure/storage/DeleteAzureStorageStep.java

```diff
@@ -0,0 +1,65 @@
+package bio.terra.workspace.service.resource.controlled.cloud.azure.storage;
+
+import bio.terra.stairway.FlightContext;
+import bio.terra.stairway.Step;
+import bio.terra.stairway.StepResult;
+import bio.terra.stairway.StepStatus;
+import bio.terra.workspace.app.configuration.external.AzureConfiguration;
+import bio.terra.workspace.service.crl.CrlService;
+import bio.terra.workspace.service.workspace.flight.WorkspaceFlightMapKeys.ControlledResourceKeys;
+import bio.terra.workspace.service.workspace.model.AzureCloudContext;
+import com.azure.resourcemanager.storage.StorageManager;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * A step for deleting a controlled Azure Storage Account resource.
+ */
+public class DeleteAzureStorageStep implements Step {
+  private static final Logger logger = LoggerFactory.getLogger(CreateAzureStorageStep.class);
+  private final AzureConfiguration azureConfig;
+  private final CrlService crlService;
+  private final ControlledAzureStorageResource resource;
+
+  public DeleteAzureStorageStep(AzureConfiguration azureConfig, CrlService crlService,
+                                ControlledAzureStorageResource resource) {
+    this.crlService = crlService;
+    this.azureConfig = azureConfig;
+    this.resource = resource;
+  }
+
+  @Override
+  public StepResult doStep(FlightContext context) throws InterruptedException {
+    final AzureCloudContext azureCloudContext =
+        context
+            .getWorkingMap()
+            .get(ControlledResourceKeys.AZURE_CLOUD_CONTEXT, AzureCloudContext.class);
+
+    StorageManager manager = crlService.getStorageManager(azureCloudContext, azureConfig);
+    var azureResourceId =
+        String.format(
+            "/subscriptions/%s/resourceGroups/%s/providers/Microsoft.Storage/storageAccounts/%s",
+            azureCloudContext.getAzureSubscriptionId(),
+            azureCloudContext.getAzureResourceGroupId(),
+            resource.getStorageAccountName());
+    try {
+      logger.info("Attempting to delete Azure Storage Account " + azureResourceId);
+      manager.storageAccounts().deleteById(azureResourceId);
```

Comment:
I followed the pattern of the other delete steps by using`deleteById`, but it looks like we could also use `deleteByResourceGroup` like we do if the [creation step fails](https://github.com/DataBiosphere/terra-workspace-manager/blob/main/service/src/main/java/bio/terra/workspace/service/resource/controlled/cloud/azure/storage/CreateAzureStorageStep.java#L103). I don't know if one is better than the other.

Reply:
`deleteByResourceGroup` might be _slightly_ cleaner so you don't have to do the string interpolation above and we can potentially reuse the deletion code between the two steps.

Reply:
I agree that `deleteByResourceGroup` seems friendlier, but again I was following the pattern of other delete steps.

@ddietterich do you know the history of using `deleteById` vs. `deleteByResourceGroup`?

Reply:
I switched to `deleteByResourceGroup` as it seems less brittle. If there is a good reason to use `deleteById`, I can change it back in my PR where I add storage containers.